### DISCUSSION TITLE
Sign-in errors redirect to /error instead of /signin ?

### DISCUSSION THREAD

**Discussion by ptrkc** [nextauthjs/next-auth#3336] [Help]
Hello everyone. I am a bit confused about errors redirect.
I have a custom sign-in page, `/login`, and by reading [the docs](https://next-auth.js.org/configuration/pages#error-codes), I understand that Sign-in page errors should redirect to `/login?error=`.
I tried to sign-in with a broken Email provider and got redirected to `/login?error=EmailSignin`, good.
Now, I tried adding a custom error page, `/error`, to handle other errors like expired sign-in link, etc.
After adding the custom error page, I no longer get redirected to `/login?error=EmailSignin`, but to `/error?error=EmailSignin`.

Is this the expected behavior? I know I can use `redirect: false` and treat the errors on the sign-in page, but I think the documentation says sign-in page errors should redirect to itself.

**Comment by balazsorban44** [comment]
shouldn't be the case. do you have a reproduction?

**Reply by ptrkc** [reply]
```javascript
export default NextAuth({
  session: { strategy: 'jwt' },
  secret,
  adapter: PrismaAdapter(prisma),
  providers: [
    EmailProvider({
      async sendVerificationRequest({ identifier: email, url }) {
        await sendEmail({ to: email, ...signInMailer({ url, email }) })
      },
    }),
  ],
  callbacks: {
    jwt: async({ token, user }) => {
      if (user) {
        return { ...token, user: { id: user.id, role: user.role } }
      }
      return token
    },
    session: async ({ session, token }) => ({ ...session, user: { id: token.user.id, role: token.user.role } }),
  },
  pages: {
    signIn: '/login',
    verifyRequest: '/verify-request',
    error: '/error',
  },
  theme: {
    logo: '/images/logo.svg',
    colorScheme: 'light',
  },
  debug: false,
})
```
and to sign-in on `/login` it's just a `signIn('email', { email })` but I've noticed it happens also with the default auth page. If I add a custom `/error` page, it get's redirected there.

**Comment by balazsorban44** [comment]
I think there are actually two errors here. The one that you describe, but it also seems to me that the current error message is shown on the built-in page just doesn't make sense:

https://github.com/nextauthjs/next-auth/blob/4dcdb62dcabad8cbc14a5357cf6c5690bb12c3d1/src/core/pages/signin.tsx#L51

This error is shown when the e-mail wasn't sent, why would we say check your inbox...

I'll convert this to an actual bug report.
